frontend/test/test-utils.tsx

```diff
@@ -181,7 +182,20 @@ export const createMockRouter = (overrides?: Partial<InjectedRouter>) => {
   };
 };
 
-export const createMockLocation = (
+export const createMockLocation = (overrides?: Partial<Location>): Location => {
+  return {
+    pathname: "/",
+    search: "",
+    hash: "",
+    query: {},
+    state: undefined,
+    action: "POP",
+    key: "",
+    ...overrides,
+  };
+};
+
+export const createMockLocationExperimental = (
```

Comment:
Is this used anywhere?

Reply:
It's not - I initially wrote it a while back as part of some feature work that didn't end up requiring it and figured leave it in there as it would be useful.I use the `Location` type imported from `"history"`here as it was more readily compatible with the `location` that `useTeamIdParam` expects, which is neither `Location` nor `IRouterLocation`, but overlaps enough with `Location` to work.

I added a discussion of all this and how we can better coordinate how we handle `location`s to the frontend sync, since it would be best for us all to be using the same types

Reply:
Sounds good. You might also be interested in the `WithRouterProps` pattern that I used on the Apple MDM SSO [pages](https://github.com/fleetdm/fleet/blob/2b57fb170aa0f9ab28a5440b8f760c3c5a9a6e39/frontend/pages/MDMAppleSSOPage/MDMAppleSSOPage.tsx#L16-L17).